ftw/subsite/viewlets/subsitelogoviewlet.py

```diff
@@ -18,19 +19,21 @@ def __init__(self, context, request, view, manager=None):
     def update(self):
         super(SubsiteLogoViewlet, self).update()
 
-        portal = self.portal_state.portal()
+        navroot = api.portal.get_navigation_root(self.context)
+        is_subsite = bool(navroot.portal_type == 'ftw.subsite.Subsite')
+
         self.navigation_root_url = self.portal_state.navigation_root_url()
 
         subsite_logo = getattr(self.context, 'logo', None)
```

Comment:
@raphael-s This does no solve the origin problem. `self.context`, needs to be replaces with `navroot`